main.tf

```diff
@@ -14,11 +14,114 @@
  * limitations under the License.
  */
 
-terraform {
-  required_version = "~> 0.12.0"
+resource "google_healthcare_dataset" "dataset" {
+    provider = google-beta
+    name = var.name
+    project = var.project
+    location = var.location
+    time_zone = var.time_zone
 }
 
-resource "google_storage_bucket" "main" {
-  project = var.project_id
-  name    = var.bucket_name
+resource "google_healthcare_dataset_iam_member" "dataset_iam_members" {
+    provider = google-beta
+    for_each = {
+        for m in var.iam_members:
+            "${m.role} ${m.member}" => m
+    }
+    dataset_id = google_healthcare_dataset.dataset.id
+    role = each.value.role
+    member = each.value.member
+}
+
+resource "google_healthcare_dicom_store" "dicom_stores" {
+    provider = google-beta
+    for_each = {
+        for s in var.dicom_stores:
+            s.name => s
+    }
+    name = each.value.name
+    dataset = google_healthcare_dataset.dataset.id
+}
+
+resource "google_healthcare_fhir_store" "fhir_stores" {
+    provider = google-beta
+    for_each = {
+        for s in var.fhir_stores:
+            s.name => s
+    }
+    name = each.value.name
+    dataset = google_healthcare_dataset.dataset.id
+}
+
+resource "google_healthcare_hl7_v2_store" "hl7_v2_stores" {
+    provider = google-beta
+    for_each = {
+        for s in var.hl7_v2_stores:
+            s.name => s
+    }
+    name = each.value.name
+    dataset = google_healthcare_dataset.dataset.id
+}
+
+locals {
+    all_dicom_iam_members = flatten([
+        for s in var.dicom_stores: [
+            for m in s.iam_members: {
+                store_name = s.name
+                role = m.role
+                member = m.member
+            }
+        ]
+    ])
+    all_fhir_iam_members = flatten([
+        for s in var.fhir_stores: [
+            for m in s.iam_members: {
+                store_name = s.name
+                role = m.role
+                member = m.member
+            }
+        ]
+    ])
+    all_hl7_v2_iam_members = flatten([
+        for s in var.hl7_v2_stores: [
+            for m in s.iam_members: {
+                store_name = s.name
+                role = m.role
+                member = m.member
+            }
+        ]
+    ])
+}
+
+resource "google_healthcare_dicom_store_iam_member" "dicom_store_iam_members" {
```

Comment:
I'd suggest we split all iam resources into an `iam.tf` file.

Reply:
Done.